### ISSUE TITLE
Observations from testing

### ISSUE THREAD

**Issue by AdamRoden** [RootKit-Org/AI-Aimbot#52]
For the curious people out there I have done some testing on this and my own aimbots using a GTX 1080ti GPU. This is about as good as you'll find without writing something yourself.

At a high-level:
- Once you get it setup properly it will work to target a humanoid's head.
- It can zero-in extremely fast on an slow moving or idle characters, but it will always lag behind running or jumping targets.
- The children I play with can aim and shoot faster and more accurately than this in 90% of scenarios.

In a deeper-dive:
- I find dxcam to be awesome! 
- I modified to use yolov8 and feel the speed and accuracy are superior.
- Using a small PyTorch model I can get 45 fps
- Using a small TensorRT model I can get over 60 fps
-

**Comment by AdamRoden** [comment]
> @AdamRoden **Using the TensortRT version ( you need to convert the model yourself ) will increase the fps a lot for me from 50 fps to 100 fps, to reduce the lag behind running players increase the speed of the mouse movement since the mouse movement is already bad, it will lag more and be jittery so you also need to find a better way to move the mouse or find a better algorithm, currently, the rookit aimbot is just for educational purposes and just a proof of concept so they didn't do these things to be plug and play cheat This can be better than memory cheats if you know how to improve it, also you need to train a better model that works on Fortnite players.**

@hafyzwithawhy 
Thanks for the reply!
In this ticket I wasn't looking for any improvements. I just wanted to tell people what they could expect from using this in the real world. 
I don't understand why you think increasing the speed of mouse movement would help anything when it is not the problem. The real problem is that the AI predicts where an object is from a still frame image, but to match what a real person can do it would need to predict where a dynamic object is going to be and move the mouse to a future coordinate, (like you said: better algorithm.)
I've been using this project to learn. I plugged dxcam into my other code and it did improve fps, but it might be causing some game lag issues. I'm still testing and learning.
I have not been able to get TensorRT to make it through the pip install, and have yet to resolve the exception it throws.
Lastly, I don't think I ever mentioned Fortnite. Why did you bring that up. Do you have a better model? I've been messing with this code and Lunar, and I want to learn more create something with YOLOv8 and train my own model.

**Comment by Qfc9** [comment]
Great stuff yall. These are great and legit analysis. 

The bot in it's current state is more geared for educational usage and is set up to be built upon. We have no plans to add or accept any PRs for adding a GUI like how Lunar has. This is because it takes away for the education value. Since we want to use this to get people interested in programming.

Anyone is free to modify the code base, add a GUI, and even sell it if they want. But in guidance with our license, credit must be given to us for the base code.

YOLOv8 was tested and can give better results, but requires beefier hardware. We have it currently in a lightweight state. Just using a medium size YOLOv5 model would give better performance in game, but requires better hardware.

**Comment by AdamRoden** [comment]
@Qfc9 I'm trying to understand how to implement yolov8 and am hung up on converting the tensor object returned by the yolo model into a useable list of numeric values, (x1, y1, x2, y2, conf, cls). It seems like this should be so simple, but I can't seem to figure it out. Could you offer some guidance?

**Comment by AdamRoden** [comment]
I'm back with more information. I haven't been using this aimbot other than as a reference for my own project to compare performance. I've found that yolov8 is significantly faster than yolov5, especially for the medium and large models. After converting the model to a TensorRT version I only gained 10-30% over the yolov8l.pt model. It knocks out 30-40 fps, but gets hammered when sharing resources with other graphics operations. Overall, I find yolo detection to be unreliable. I would need to invest significant time to train a custom model.

It's been a fun learning experience. Even with aim assistance I still suck and shooters.
